### ISSUE TITLE
False-positive compiler warning about enumeration value ‘binary’ not handled in switch

### ISSUE THREAD

**Issue by kernie** [pboettch/json-schema-validator#169]
Hi,

compiling your project with g++ versions from 7 to 10  I get following warning:
```
json-schema-validator/src/json-validator.cpp: In static member function ‘static std::shared_ptr<{anonymous}::schema> {anonymous}::type_schema::make(nlohmann::json&, nlohmann::basic_json<>::value_t, nlohmann::json_schema::root_schema*, const std::vector<nlohmann::json_uri>&, std::set<std::__cxx11::basic_string<char> >&)’:
/home/id6945/Development/smure/source/thirdparty/json-schema-validator/src/json-validator.cpp:1098:9: warning: enumeration value ‘binary’ not handled in switch [-Wswitch]
 1098 |  switch (type) {
      |         ^
```
I checked json.hpp and in your source code you handle all values of json::value_t, so I think this is a false warning. However, when using strict compilation, this is in the way...

One can avoid this warning e.g. with a default switch followed by a break as action.

Not sure what is the best way to clean the warning.

Kind regards

**Comment by pboettch** [comment]
Which version are you using?

There is this in current upstream: https://github.com/pboettch/json-schema-validator/blob/master/src/json-validator.cpp#L1218

**Comment by kernie** [comment]
2.1.0

**Comment by pboettch** [comment]
OK, yes. This warning is fixed upstream on the master-branch.
